contracts/RootChain.sol

```diff
@@ -1094,24 +1124,56 @@ contract RootChain {
      * @param _tx RLP encoded transaction.
      * @return A tuple containing the number of inputs and the sum of the outputs of tx.
      */
-    function _getOutputInfo(bytes _tx)
+    function _validateInputsOutputsSumUp(InFlightExit storage _inFlightExit, bytes _tx)
         internal
         view
-        returns (uint8, uint256)
     {
+
+        InputSum[MAX_INPUTS] memory sums;
+        uint8 allocatedSums = 0;
+
+        InputSum memory tokenSum;
+
+        uint8 i;
         // Loop through each input.
-        uint8 numInputs = 0;
-        uint256 outputSum = 0;
-        for (uint8 i = 0; i < MAX_INPUTS; i++) {
-            if (_tx.getInputUtxoPosition(i) > 0) {
-                numInputs++;
+        for (i = 0; i < MAX_INPUTS; ++i) {
+            PlasmaCore.TransactionOutput memory input = _inFlightExit.inputs[i];
+            (tokenSum, allocatedSums) = _getInputSumByToken(sums, input.token, allocatedSums);
+
+            tokenSum.amount += input.amount;
+        }
+
+        for (i = 0; i < MAX_INPUTS; ++i) {
+            PlasmaCore.TransactionOutput memory output = _tx.getOutput(i);
+            (tokenSum, allocatedSums) = _getInputSumByToken(sums, output.token, allocatedSums);
+
+            require(tokenSum.amount >= output.amount);
+            tokenSum.amount -= output.amount;
+        }
```

Comment:
This function needs more comments. At least for all things that are both implicit and critical. E.g. underflow protection and reuse of memory for sums.